semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TreeOps.scala

```diff
@@ -0,0 +1,44 @@
+package scala.meta.internal.semanticdb.scalac
+
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.inputs._
+
+trait TreeOps { self: SemanticdbOps =>
+  import g._
+
+  implicit class XtensionGTreeSyntheticTerm(gTree: g.Tree) {
+
+    def toSemanticTree: s.Tree = gTree match {
+      case gTree: g.Apply =>
+        s.ApplyTree(
+          fn = gTree.fun.toSemanticId,
+          args = gTree.args.map(_.toSemanticTree)
+        )
+      case gTree: g.TypeApply =>
+        s.TypeApplyTree(
+          fn = gTree.fun.toSemanticTree,
+          targs = gTree.args.map(_.tpe.toSemanticTpe)
+        )
+      case gTree: g.Select => gTree.toSemanticId
+      case gTree: g.Ident => gTree.toSemanticId
+      case gTree: g.This => gTree.toSemanticId
+      case gTree: g.Typed =>
+        s.MacroExpansionTree(
+          expandee = gTree.expr.toSemanticTree,
+          tpe = gTree.tpt.tpe.toSemanticTpe
+        )
+      case _ =>
+        println(s"No match on: $gTree ${gTree.getClass}")
```

Comment:
Also, I don't think we should be issuing unpositioned and unconditional messages in production. I suggest we remove this println unless we're willing to introduce dedicated configuration option for this (personally, I don't think that's worth it).